src/Popper.js

```diff
@@ -59,24 +57,37 @@ const initialArrowStyle = {};
 
 export class InnerPopper extends React.Component<PopperProps, PopperState> {
   static defaultProps = {
+    init: true,
```

Comment:
I've added this so implementors can lazy initialize popper, such as in the case of tooltip or dropdown, only when the popper element is visible

Reply:
May you help me to understand better? If I set this boolean to `true` and then to `false, what happens?

Reply:
Also, is there any reason to prefer this rather than not making React render the whole component completely?

I mean:

```jsx
init = false;
[...]
{init && <Tooltip />}
```

rather than:
```jsx
<Tooltip init={false} />
```

Reply:
Yeah I'm not set on this API yet, I'm still testing it out in terms of what we need for react-bootstrap.

Generally yes conditionally rendering the popper element would handle this case and honestly I think that's the best approach. I've added it to enable the bootstrap behavior of keeping drop-down menus rendered even when closed (initially). We could render the menu without the popper element intially but that would cause the drop-down to unmount and remount can update when it opens, since it's parent is changing

The other thing I wanted to enable is clearer control on when popper is initiated vs updated. I don't really think this API solves that well but it was my first attempt, and maybe schedule update is enough.

Let me play around with this a bit more

Reply:
wrote the above on my phone, but here is an example for why a toggle prop may be preferable

```js
{show ? <Popper>{() => <DropdownMenu />}</Popper> : <DropdownMenu />}
```

if it is required that the menu always be rendered the above works but causes a remount on each `show` toggle vs staying mounted but updating.

Reply:
I guess what I'm trying to understand is in which case you want to switch from a positioned to not-positioned version of the same dropdown. I understand the Bootstrap case, but there's not dynamic switching involved there, there are some places where Popper.js is used and places where it's not, that's it, right?

Reply:
I don't usually want to do switch between positioned and not, I _do_ want to defer initializing popper until a user opens a tooltip or dropdown, so all the popper init work isn't done when the page loads, but no tooltips or dropdowns are visible.

The actual common case here is something like this: 

```js
{hasShownOnce ? <Popper>{() => <DropdownMenu />}</Popper> : <DropdownMenu />}
```

so once the dropdown is shown it, popper stays, but not until the first `show`. I will say tho that vanilla bootstrap does recreate `popper` on each show, i'm not sure why they do that vs doing a scheduleUpdate tho.

Reply:
It may also be worth not running updates initiated from `onScroll` or window resize when the element isn't visible? Maybe popper already optimizes this?

Reply:
Umh okay... Last question (I hope), why do you need to render the dropdown even when you don't want to show it? Couldn't it be:

Reply:
Also, now that I think about it, Popper.js accepts a `eventsEnabled` option which may be used in place of this custom implementation maybe?

Reply:
In any case, maybe it would make sense to move this to a different PR and leave this one focused on perf improvement so that it's easier to review it, if it's a problem I can do it myself.

Reply:
> Couldn't it be:
` {hasShownOnce && <Popper>{() => <DropdownMenu />}</Popper>}`

They only reason i'm a bit concerned with doing that, is that it's different from how bootstrap and react-bootstrap handle dropdowns by default. Given how many users RB has, i'm not confident that not mounting the DropdownMenu initially won't break someone's workflow. People put a ton of logic in their components and it's not clear that they won't need that to run immediately (maybe the dropdown item needs to attach global key handlers for keycode shortcuts i don't know).

I'm happy to try the `eventsEnabled` as an initial strategy, it's easy to add this API in the future if it's needed, and harder to remove it if it's not